mage_ai/data_preparation/models/block/__init__.py

```diff
@@ -2111,6 +2125,36 @@ def update(self, data, **kwargs) -> 'Block':
                 check_upstream_block_order=check_upstream_block_order,
             )
 
+        if 'downstream_blocks' in data and self.pipeline:
```

Comment:
Same here I would add this logic to to `pipeline.update_block` method so that we can validate the pipeline after updating the dependency.

Reply:
Done.